File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.util.Scanner;

public class Main
{
    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);
        String[] array = scanner.nextLine().split("\\s");
        int hour = Integer.parseInt(array[0]);
        int minute = Integer.parseInt(array[1]);

        if (minute >= 45)
        {
            System.out.printf("%s %s", Integer.toString(hour), Integer.toString(minute - 45));
        }
        else
        {
            if (hour > 0)
            {
                System.out.printf("%s %s", Integer.toString(hour - 1), Integer.toString(minute + 15));
            }
            else
            {
                System.out.printf("%s %s", Integer.toString(hour + 23), Integer.toString(minute + 15));
            }
        }
    }
}

File: 백준/Bronze/2884. 알람 시계/알람 시계.py
hour, minute = map(int, input().split())

minute_tmp = minute - 45

if minute_tmp >= 0:
    minute = minute_tmp
else:
    minute = minute_tmp + 60
    hour_tmp = hour - 1
    if hour_tmp >= 0:
        hour = hour_tmp
    else:
        hour = 24 + hour_tmp
    
print(f'{hour} {minute}')